hyperspy/drawing/utils.py

```diff
@@ -343,6 +343,466 @@ def _set_spectrum_xlabel(spectrum, ax):
+        if len(label_list) < n:
+            label_list *= (n / len(label_list)) + 1
+            del label_list[n:]
+
+    else:
+        # catch all others to revert to default if bad input
+        print "Did not understand input of labels. Defaulting to image titles."
+        label_list = [x.metadata.General.title for x in images]
+
+    # Determine appropriate number of images per row
+    rows = int(np.ceil(n / float(per_row)))
+    if n < per_row:
+        per_row = n
+
+    # Set overall figure size and define figure (if not pre-existing)
+    if fig is None:
+        k = max(plt.rcParams['figure.figsize']) / max(per_row, rows)
+        f = plt.figure(figsize=(tuple(k * i for i in (per_row, rows))))
+    else:
+        f = fig
+
+    # Initialize list to hold subplot axes
+    axes_list = []
+
+    # Initialize list of rgb tags
+    isrgb = [False] * len(images)
+
+    # Check to see if there are any rgb images in list
+    # and tag them using the isrgb list
+    for i, img in enumerate(images):
+        if rgb_tools.is_rgbx(img.data):
+            isrgb[i] = True
+
+    # Find global min and max values of all the non-rgb images for use with 'single' scalebar
+    if colorbar is 'single':
+        gl_max, gl_min = max([i.data.max() for i in list(compress(images, [not j for j in isrgb]))]), \
+                         min([i.data.min() for i in list(compress(images, [not j for j in isrgb]))])
+
+    # Check if we need to add a scalebar for some of the images
+    if isinstance(scalebar, list) and all(isinstance(x, int) for x in scalebar):
+        scalelist = True
+    else:
+        scalelist = False
+
+    idx = 0
+    # Loop through each image, adding subplot for each one
+    for i, ims in enumerate(images):
+        # Get handles for the signal axes and axes_manager
+        axes_manager = ims.axes_manager
+        axes = axes_manager.signal_axes
+        if axes_manager.navigation_dimension > 0:
+            ims = ims._deepcopy_with_new_data(ims.data)
+        for j, im in enumerate(ims):
+            idx += 1
+            ax = f.add_subplot(rows, per_row, idx)
+            axes_list.append(ax)
+            data = im.data
+
+            # Enable RGB plotting
+            if rgb_tools.is_rgbx(data):
+                data = rgb_tools.rgbx2regular_array(data, plot_friendly=True)
+            else:
+                data = im.data
+
+            # Remove NaNs (if requested)
+            if no_nans:
+                data = np.nan_to_num(data)
+
+            # Get handles for the signal axes and axes_manager
+            axes_manager = im.axes_manager
+            axes = axes_manager.signal_axes
+
+            if axes_manager.signal_dimension == 2:
+                # get calibration from a passed axes_manager
+                shape = axes_manager._signal_shape_in_array
+
+                # Reshape the data for input into imshow (if not rgb)
+                if not isrgb[i]:
+                    data = data.reshape(shape)
+
```

Comment:
Hmm, I suppose you're right. It works fine without it. I'm not sure how/why that was in there, actually... I may have just copied it from the `signal._plot_loadings()` function at the beginning.